src/NetMQ/NetMQSocket.cs

```diff
@@ -22,6 +22,14 @@ public abstract class NetMQSocket : IOutgoingSocket, IReceivingSocket, ISocketPo
         private EventHandler<NetMQSocketEventArgs> m_sendReady;
         private bool m_isClosed;
 
+        /// <summary>
+        /// The Socket Class type identifier for this Socket
+        /// </summary>
+        public abstract ZmqSocketType SocketType
+        {
+            get;
+        }
+
```

Comment:
Thanks, my oversight; didn't check well enough beforehand.  I'll refactor
it but likely will delete the reauest and wait until I have something a bit
more meaningful before trying again.

This request was trying to be a step towards isolating Sockets so they can
be inplemented using their own separate assembly modules without modifying
or recompiling the NetMQ module; same goal for transports too.

Thanks!
On Jan 25, 2016 12:50 AM, "Doron Somech" <notifications@github.com> wrote: